### PR TITLE
I18n Support for Angular Authentication Components

### PR DESCRIPTION
**Pull request by seang96**
Description of changes:
Fixed build errors when trying to build on Windows (needed to do this for local development)
Added I18n inside provider AmplifyService
Added I18n support to aws-amplify-angular authentication components
Removed error message being an alert for ionic components
Made error message boxes the same throughout authentication components and in the correct location inside the template. Some were inside the amplify-form-body.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sprucify** [comment]
Thanks so much for creating this feature. I18n is essential!

File: packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts [hunk 1]
```diff
@@ -12,30 +12,31 @@
  * and limitations under the License.
  */
 // tslint:enable
+import { I18n } from '@angular/core';
```
Comment (sprucify) [inline packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: Does @angular/core have a I18n module?

Comment (seang96) [reply packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: @sprucify  @angular/core does have an existing I18n module. This has already been done to the ract/vue components and has a docs page! [https://aws-amplify.github.io/docs/js/i18n](https://aws-amplify.github.io/docs/js/i18n)

Comment (sprucify) [reply packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: Yes, but not from package @angular/core?

Comment (seang96) [reply packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: @sprucify Oh I did misread the import at first and put it in @angular/core, however I changed the imports to the correct ones afterwards. See the final commit of the file: [https://github.com/aws-amplify/amplify-js/blob/fdfd2122b5b14d2eb08d52ab31fc8caf9fbab935/packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts](https://github.com/aws-amplify/amplify-js/blob/fdfd2122b5b14d2eb08d52ab31fc8caf9fbab935/packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts)

Comment (sprucify) [reply packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: Ah sorry, I overlooked your more recent commit.
I’m going to implement this in my app and hopefully it will be part of master soon

**Comment by haverchuck** [comment]
@seang96  - Thanks so much for your PR.  This is definitely a needed addition.  It is being reviewed now - comments will be left in line.

File: packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts (hunk 1 above)
Comment (sprucify) [reply packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: I couldn’t get it working by replacing the version numbering with your repository url. (Inexperience me)
I guess I have to wait for the amplify update.

File: packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts [hunk 2]
```diff
@@ -12,30 +12,33 @@
...
+    label: 'Password *',
+    placeholder: 'Password',
     key: 'password',
     required: true,
     type: 'password',
     displayOrder: 2,
   },
   {
-    label: 'Email',
+    label: 'Email *',
```
Comment (haverchuck) [inline packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: @seang96  -  Please revert the change to these label values.  They should not have an asterix.  The required attribute should conditionally insert an asterix.  (This will be another change to remove.)

File: packages/aws-amplify-angular/src/components/authenticator/sign-up-component/sign-up.component.core.ts [hunk 3]
```diff
@@ -24,45 +24,43 @@ const template = `
...
     <div class="amplify-form-body">
-      <div class="amplify-form-header">{{this.header}}</div>
+      <div class="amplify-form-header">{{ this.amplifyService.i18n().get(this.header) }}</div>
 
       <div class="amplify-form-row" *ngFor="let field of signUpFields">
         <div *ngIf="field.key !== 'phone_number'">
           <label class="amplify-input-label">
-            {{field.label}} 
-            <span *ngIf="field.required">*</span>
+            {{ this.amplifyService.i18n().get(field.label) }}
```
Comment (haverchuck) [inline packages/aws-amplify-angular/src/components/authenticator/sign-up-component/sign-up.component.core.ts]: @seang96 Please reenable `<span *ngIf="field.required">*</span>`.  This will automatically insert an asterix if the field is required.

File: packages/aws-amplify-angular/src/components/authenticator/sign-up-component/sign-up.component.core.ts [hunk 4]
```diff
@@ -24,45 +24,43 @@ const template = `
...
             [(ngModel)]="user[field.key]" name="field.key" />
             <div *ngIf="field.key === 'password'" class="amplify-form-extra-details">{{passwordPolicy}}</div>
         </div>
-            
+
         <div *ngIf="field.key === 'phone_number'">
           <label class="amplify-input-label">
-            {{field.label}} 
-            <span *ngIf="field.required">*</span>
+            {{ this.amplifyService.i18n().get(field.label) }}
```
Comment (haverchuck) [inline packages/aws-amplify-angular/src/components/authenticator/sign-up-component/sign-up.component.core.ts]: @seang96 Please see above.

File: packages/aws-amplify-angular/src/components/authenticator/sign-up-component/sign-up.component.ionic.ts [hunk 5]
```diff
@@ -39,70 +38,67 @@ const template = `
...
                 <ion-label class="amplify-input-label push-right">&nbsp;</ion-label>
-                <ion-input 
+                <ion-input
                   #phone_number
                   [ngClass]="{'amplify-input-invalid ': field.invalid}"
                   *ngIf="field.key === 'phone_number'"
                   type={{field.type}}
                   class="amplify-form-input-phone-ionic"
-                  placeholder={{field.label}}
+                  [placeholder]="this.amplifyService.i18n().get(field.placeholder)"
```
Comment (haverchuck) [inline packages/aws-amplify-angular/src/components/authenticator/sign-up-component/sign-up.component.ionic.ts]: @seang96 Good catch, thank you!

File: packages/pubsub/package.json [hunk 6]
```diff
@@ -11,7 +11,7 @@
   "scripts": {
     "test": "tslint 'src/**/*.ts' && jest --coverage",
     "build-with-test": "npm run clean && npm test && tsc && webpack",
-    "build": "npm run clean && tsc && cp -r src/vendor lib/vendor && webpack",
+    "build": "npm run clean && tsc && cpx src/vendor/** lib/vendor && webpack",
```
Comment (haverchuck) [inline packages/pubsub/package.json]: Please remove these changes to PubSub package.json. If needed, please submit them in a different PR.

File: packages/pubsub/package.json [hunk 7]
```diff
@@ -35,6 +35,7 @@
     "babel-preset-react": "^6.24.1",
     "babel-preset-stage-2": "^6.24.1",
     "compression-webpack-plugin": "^1.1.3",
+    "cpx": "^1.5.0",
```
Comment (haverchuck) [inline packages/pubsub/package.json]: @seang96 Please see above.

File: packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts (hunk 2 above)
Comment (seang96) [reply packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: @haverchuck I explicitly defined these with an asterisk if they were required rather than the span element containing the element in case another language didn't use asterisks for field requirements or it was a different location than after said label.

> Not all countries use asterisks or bold type to denote a required field. For example the “※” symbol is a popular in Japan to show that something is important or deserves attention. An asterisk works as well, but “※” is so particular to Japan that it could be used to build consumer confidence so customers know they are dealing with a local company (or at least one that researched the market).

[Source](https://www.cleverbridge.com/corporate/3-tips-to-localize-your-form-fields/)

File: packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts (hunk 1 above)
Comment (seang96) [reply packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: Put this in your package.json script section, npm install to install first time, updates to branch use npm run update-amplify-js.
```
    "preinstall": "git clone https://github.com/seang96/amplify-js.git --single-branch --branch angular-auth-I18n && npm run build-amplify-js",
    "update-amplify-js": "cd amplify-js && git fetch && git reset --hard origin/angular-auth-I18n && git clean -df && cd .. && npm run build-amplify-js",
    "build-amplify-js": "cd amplify-js && yarn config set workspaces-experimental true && yarn && npx yarn bootstrap && yarn build",
```

File: packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts (hunk 2 above)
Comment (haverchuck) [reply packages/aws-amplify-angular/src/assets/default-sign-up-fields.ts]: @seang96 Ok, that makes sense.  Thanks.

**Comment by seang96** [comment]
@haverchuck I removed the package.json modifications and put them in another PR as requested.

I also just added a commit that fixed the phone number section in the sign up form for Ionic that would make it scroll on small screens. I do not think this was an issue caused by my changes, but I updated just to make sure.

**Comment by haverchuck** [comment]
@seang96 I thought about your point that the asterix does not universally denote a required field, and I think it makes sense.  However, I think we should retain it for the time being since a) our other framework packages contain this, and b) it would be considered a breaking change (albeit very minor breakage).

Removing the asterix would make sense as part of a major version change.

**Comment by seang96** [comment]
> @seang96 I thought about your point that the asterix does not universally denote a required field, and I think it makes sense. However, I think we should retain it for the time being since a) our other framework packages contain this, and b) it would be considered a breaking change (albeit very minor breakage).
> 
> Removing the asterix would make sense as part of a major version change, across framework packages. Your other changes are great and I would love to merge them ASAP.

I can agree with that, I did a new commit to revert back to the old way for those fields.
